src/client/components/equalizer/equalizer.js

```diff
@@ -1,73 +1,53 @@
 import React, { Component } from 'react';
-import Hark from 'hark';
+import Pizzicato from 'pizzicato';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import Graphicequaliser from './canvasEqualizer';
 import PlayButton from './playButton';
 import Uploadbutton from './upload';
 import Infoabouttrack from './upload/infoAboutFile';
 import Streambutton from './streamButton';
-import StreamDetect from './streamDetect';
 import './equalizer.css';
 import {
   createAudioData,
   playPauseSoundFromFile,
   createStreamData,
   startMuteStreamAudio,
-  mergeCanvasWidth,
 } from '../../actions/audioActions';
 
 class Equalizer extends Component {
   state = {
     analyser: null,
     ctx: null,
     numPoints: null,
-    heightArray: null,
-    streamDetect: 'no speaking',
+    uint8Array: null,
   }
 
   componentDidMount() {
     this.detectStreamSoundFromMicrophone();
-    const canvasEl = document.querySelector('canvas').getContext('2d');
-    const { audioData } = this.props;
+  }
+
+  static getDerivedStateFromProps(nextProps) {
+    const { audioData } = nextProps;
```

Comment:
You can destruct here analyzer and audio context if you don’t use audioData anywhere

Reply:
done